### PR TITLE
Refactor workloadentry controller 

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

To use the generic queue instead of k8s lib.

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go [hunk 1]
```diff
@@ -143,7 +143,8 @@ func (d *DeploymentController) Run(stop <-chan struct{}) {
 }
 
 // Reconcile takes in the name of a Gateway and ensures the cluster is in the desired state
-func (d *DeploymentController) Reconcile(req types.NamespacedName) error {
```
Comment (ramaraochavali) [inline pilot/pkg/config/kube/gateway/deploymentcontroller.go:146]: Why this change? Is there a problem if we leave it to have types/NamespacedName as earlier?

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:146]: The function is the workfunc of the queue, it is not always of NamespacedName type. You can see the workloadentry controller

Comment (costinm) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:146]: I looked - it seems Add() is adding a NamespacedName(), same for AddObject() -  I assume the object is retrieved is the same type - not sure why the Queue implementation uses interface{} in the function, maybe we should fix it ? 

But probably not a high priority, I can live with it as interface{}

**Review by costinm — APPROVED** [review]

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:146]: I think it is to provide high flexibility. In most cases even for most k8s controllers name/ns is enough, but for some other cases, the item can be any customized struct.
